static_src/stores/app_store.js

```diff
@@ -107,8 +110,19 @@ class AppStore extends BaseStore {
         break;
       }
 
+      case appActionTypes.APP_START: {
+        const app = this.get(action.appGuid);
+        if (app) {
+          const startingApp = Object.assign({}, app,
+            { state: appStates.starting });
```

Comment:
btw, since we're using ES6, this could be written as `startingApp = { ...app, state: appStates.starting }`, do you have an opinion on if we want to use the `Object.assign` syntax or spread operator?